packages/twenty-server/src/workspace/messaging/services/messaging-utils.service.ts

```diff
@@ -112,38 +111,53 @@ export class MessagingUtilsService {
         connectedAccount.handle === fromHandle ? 'outgoing' : 'incoming';
 
       await workspaceDataSource?.transaction(async (manager) => {
-        await manager.query(
-          `INSERT INTO ${dataSourceMetadata.schema}."message" ("id", "externalId", "headerMessageId", "subject", "receivedAt", "messageThreadId", "direction", "body") VALUES ($1, $2, $3, $4, $5, $6, $7, $8)`,
-          [
-            messageId,
-            externalId,
-            headerMessageId,
-            subject,
-            receivedAt,
-            messageThread[0]?.id,
-            messageDirection,
-            text,
-          ],
+        const existingMessages = await manager.query(
+          `SELECT * FROM ${dataSourceMetadata.schema}."message" WHERE ${dataSourceMetadata.schema}."message"."headerMessageId" = $1`,
+          [headerMessageId],
         );
```

Comment:
If the message already exists, we are not sure that 'bcc' recipients are already imported since only the person sending the email will see the whole list of 'bcc' and a person who is in bcc will only see its email address as a bcc.